src/docs/arch-cursor.dox

```diff
@@ -371,6 +371,15 @@ how many references to closed data handles are freed, we may continue our walk.
 usually strike a good balance between not having a lot of overhead for sweeps, and keeping up with
 the need to free up shared resources.
 
+@section arch_cursor_bound Range bounded cursors
+
```

Comment:
@ddanderson does this seem like the right level of detail for the architecture guide?

Reply:
Yes, I think it's at the right level.  When I'm writing/reviewing these sections, I'm thinking about:

- a generic overview of the code (and not a rehash of user documentation), it can be helpful to have the overview before someone dives into one piece of the code.  Often this is a quick discussion of any internal data structure, how it is used, and any "rules" that must be followed when using the data structure.
- somewhat equivalently, those things that an experienced coder might learn from a quick scan of the code.
- any other subtle things that it's hard to see by looking at the code.

Many of the above can/should also appear in code comments.